apps/app/src/test/resources/total-supply-bigquery.sql

```diff
@@ -149,37 +157,57 @@ CREATE TEMP FUNCTION unminted(
     migration_id));
 
 
-CREATE TEMP FUNCTION TransferResult_summary(summary_position int64)
-    RETURNS string AS (
-  daml_record_path([1, summary_position], 'numeric'));
+CREATE TEMP FUNCTION TransferSummary_minted(tr_json json)
+    RETURNS bignumeric AS (
+  daml_record_numeric(tr_json, [0]) -- .inputAppRewardAmount
+  + daml_record_numeric(tr_json, [1]) -- .inputValidatorRewardAmount
+  + daml_record_numeric(tr_json, [2]) -- .inputSvRewardAmount
+);
 
+-- A choice's TransferSummary, if that choice is filtered by `minted`.
+CREATE TEMP FUNCTION choice_result_TransferSummary(choice string, result json)
+    RETURNS json AS (
+  CASE choice
+    WHEN 'AmuletRules_CreateExternalPartySetupProposal'
+      -- .transferResult.summary
+      THEN JSON_QUERY(result, daml_record_path([3, 1], 'record'))
+    WHEN 'AmuletRules_CreateTransferPreapproval'
+      -- .transferResult.summary
+      THEN JSON_QUERY(result, daml_record_path([1, 1], 'record'))
+    WHEN 'AmuletRules_BuyMemberTraffic'
+      -- .summary
+      THEN JSON_QUERY(result, daml_record_path([1], 'record'))
+    WHEN 'AmuletRules_Transfer'
+      -- .summary
+      THEN JSON_QUERY(result, daml_record_path([1], 'record'))
+    WHEN 'TransferPreapproval_Renew'
+      -- .transferResult.summary
+      THEN JSON_QUERY(result, daml_record_path([1, 1], 'record'))
```

Comment:
`daml_record_path([1, 1], 'record')` counts as a "constant" argument for `JSON_QUERY` according to BQ. Doing the obvious factoring...does not. 😒